test/support.ts

```diff
@@ -20,6 +20,27 @@ export async function submit(
   return receipt;
 }
 
+/// Submit a transaction and expect it to fail. Throws an error if it succeeds.
+export async function submitFailing(
+  tx: Promise<Ethers.ContractTransaction>
+): Promise<void> {
+  let succeeded = false;
+
+  try {
+    const receipt = await (await tx).wait();
+
+    if (receipt.status == 1) {
+      succeeded = true;
+    }
+  } catch {
+    /* OK */
```

Comment:
I don’t think we should ignore exceptions here. If one is raised this means that the transaction was not included in a block which is probably not what we want. If you remove the `try/catch` block then the tests should still pass but we will also be alerted when something is wrong with submitting the transcation.

Reply:
But it's inside `submitFailing`, it's supposed to revert

Reply:
If I remove the try/catch, the test will fail, because an uncaught exception will be raised.